src/scripts/ordo_subsets.py

```diff
@@ -0,0 +1,83 @@
+"""Creates a ROBOT template to update mondo terms with the correct ORDO subset annotation."""
+from argparse import ArgumentParser
+
+import pandas as pd
+
+from utils import remove_angle_brackets
+
+
+SUBSET_MAP = {
+    'Orphanet:557493': 'http://purl.obolibrary.org/obo/mondo#ordo_disease',  # disorder
+    'Orphanet:557492': 'http://purl.obolibrary.org/obo/mondo#ordo_group_of_disorders',  # group of disorders
+    'Orphanet:557494': 'http://purl.obolibrary.org/obo/mondo#ordo_clinical_subtype',  # subtype of a disorder
+}
+ROBOT_TEMPLATE_HEADER = {
+    'mondo_id': 'ID',
+    'ordo_id': 'A oboInOwl:hasDbXref',
+    'subset': 'A oboInOwl:inSubset',
+}
+
+
+def create_ordo_subsets_robot_template(
+    class_subsets_tsv_path: str, mondo_mappings_path: str, outpath: str,
+):
+    """Creates a ROBOT template to update mondo terms with the correct ORDO subset annotation."""
+    # Read: mondo.sssom.tsv
+    mondo_df = pd.read_csv(mondo_mappings_path, sep='\t', comment='#').rename(columns={
+        'subject_id': 'mondo_id',
+        'subject_label': 'mondo_label',
+    })
+    # - keep only exact matches
+    mondo_df = mondo_df[mondo_df['predicate_id'] == 'skos:exactMatch']
+
+    # Read: ORDO class subsets query results
+    df = pd.read_csv(class_subsets_tsv_path, sep='\t').rename(columns={
+        '?cls': 'ordo_id',
+        '?lbl': 'ordo_label',
+        '?grp': 'subset_ordo_class_id',
+        '?grp_lbl': 'subset_ordo_class_label',
+    }).sort_values(['ordo_id', 'ordo_label'])
+    # - remove duplicate rows where labels are a specific language variation, e.g. @en
+    #   this alternative works as of 2024/05/02, but is potentially trepidacious:
+    # df2 = df.drop_duplicates(subset=['ordo_id'], keep='last')
+    df['label_has_lang_variation'] = df['ordo_label'].map(lambda label: '@' in label)
+    df = df[~df['label_has_lang_variation']]
+    del df['label_has_lang_variation']  # this gets dropped later, but w/e
+
+    # Preprocess
+    for fld in ['ordo_id', 'subset_ordo_class_id']:
+        # - Strip annoying angle brackets from URIs
+        df[fld] = df[fld].map(remove_angle_brackets)
+        # - Compress class IDs
+        df[fld] = df[fld].map(lambda x: x.replace('http://www.orpha.net/ORDO/Orphanet_', 'Orphanet:'))
```

Comment:
## Temporary lack of `curies`
Due to lack of time. I can update this pre or post merge.

Reply:
What needs updating here?

Reply:
If the outputs of this script are good and we never need to run this again, then I don't think we strictly need to do anything here.

But if we want this code to be standardized, then instead of this find/replace I'm doing, I would pass the path to one of the `mondo-ingest` config files which contains `curie_map`, and then I would use the `curies` package here, something like fist instantiate a `converter` and then `lambda x: converter.collapse(x)`.

Reply:
If it's not going to be used again, then we technically don't even have to merge it. But I think it's good to commit these ad hoc workflows somewhere.

Maybe we just don't know if we'll need to use this code again, or maybe we might borrow this code to do something similar, in which case we ideally do want the code to be high / standard quality.

That being said, it should take 5-10 minutes to update this properly.

Reply:
This code _will definitely be used again_, as the ORDO subset pipeline is integral now to how we identify what Orphanet considers a rare disease. This pipeline will be run basically once every month!

I did not clearly express myself when I asked for this issue: I wanted a quick and dirty solution (one off) so I can implement this pipeline before the next release, and give you the time to develop a proper solution in the future. However, you are already so close to a proper solution that I would just suggest to push it over the finish line now.

Reply:
To state here the Slack comment, Pause on making this the proper solution for now.

Reply:
Resolving this conversation as this will be handled in:
- #515